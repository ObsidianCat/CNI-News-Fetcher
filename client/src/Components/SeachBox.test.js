import React from 'react';
import { mount } from 'enzyme';
import SearchBox from "./SeachBox";

let wrapper;
const fetchNewsMock = jest.fn();

beforeEach(()=>{
  wrapper =  mount(<SearchBox fetchNews={fetchNewsMock} />);
})

afterEach(()=>{
  wrapper.unmount()
})

it('renders as expected', () => {
  expect(wrapper).toMatchSnapshot();
});

it('calls fetchNews function on form submit', () => {
  wrapper.setState({ term: 'Sunday' });
  wrapper.find('form').simulate('submit')

  expect(fetchNewsMock).toBeCalledWith('/news/filtered?query=Sunday');
});