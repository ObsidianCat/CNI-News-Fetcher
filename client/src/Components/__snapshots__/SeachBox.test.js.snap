// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<SearchBox
  fetchNews={[MockFunction]}
>
  <form
    className="search-box input-group"
    onSubmit={[Function]}
  >
    <Input
      className="search"
      disabled={false}
      onChange={[Function]}
      placeholder="type something to search"
      prefixCls="ant-input"
      type="text"
      value=""
    >
      <input
        className="ant-input search"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="type something to search"
        type="text"
        value=""
      />
    </Input>
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      prefixCls="ant-btn"
      type="primary"
    >
      <Wave>
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="submit"
        >
          <span
            key=".0"
          >
            Submit
          </span>
        </button>
      </Wave>
    </Button>
  </form>
</SearchBox>
`;
